rust/core-lib/src/editor.rs

```diff
@@ -1058,6 +1058,17 @@ impl<W: Write + Send + 'static> Editor<W> {
         Some(text.slice_to_string(offset, end_off))
     }
 
+    pub fn plugin_get_selections(&self, view_id: &ViewIdentifier) -> Value {
+        //TODO: multiview support
```

Comment:
Yah I think that would make a lot of sense, instead of things running on "the main view of this editor", they would run on "this view". 

Sublime's entire API is on the view and it just happens that internally that views sometimes share buffers, and they give you a way to get the buffer ID of a view so you can tell which views share buffers.

Alternatively, there could be some things that run on views and some that run on buffers and there is a pool of each that you index with a buffer or view ID. There's APIs to get the buffer ID of a view or the views displaying a buffer. This I think is how Emacs works. This API more explicitly represents the underlying concepts and also enables buffers that aren't attached to views, on the other hand it is more complex, and in most cases you don't care and the Sublime-style API just does what you want. You also need some sort of garbage collector, or you can do what Emacs does and have your number of open buffers just gradually grow forever unless you intervene or install a garbage collector plugin 😛 .

Interestingly, the internal representation can be different than what the plugin API makes things look like. In fact I think I might be in favor of exposing a Sublime-style API to plugins but using two pools internally.

I have a feeling that there exists an architecture with two pools that is nice and clean, but I don't actually know what it would look like or how to go about doing that. On the other hand I think making buffers shared reference counted children of views is more straightforward.